### PR TITLE
Improved test coverage

### PR DESCRIPTION
**Pull request by lustefaniak**

### CODE REVIEW

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  test("Allows to get archive between two Actors") {
+    val arch = new ArchiveSupport {
+      override def isArchiveEnabled: Boolean = true
+
+      override def receivedBetween(ref: String, ref2: String): Source[ReceivedRecord, _] =
+        Source.single(ReceivedRecord(uuid, 200, ref, akkaviz.persistence.To, ref2, "{}"))
+
+      override def receivedOf(ref: String): Source[ReceivedRecord, _] = ???
```
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328473506/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  val uuid = UUID.randomUUID()
+
+  val ofGet = Get("/messages/of/ref")
+  val betweenGet = Get("/messages/between/ref/ref2")
+
+  test("Allows to get archive of Actor") {
+    val arch = new ArchiveSupport {
+      override def isArchiveEnabled: Boolean = true
+
+      override def receivedBetween(ref: String, ref2: String): Source[ReceivedRecord, _] = ???
```
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328473506/issues/source?bid=3121282&fileBranchId=3206949#l21)

Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328538975/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328538975/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328652494/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328652494/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328672076/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328672076/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328681403/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328681403/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328728990/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328728990/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328850840/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2328850840/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329236807/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329236807/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    val wsRepl = new WebSocketRepl {
+      override protected def startReplProcessing(
+        in: InputStream, out: OutputStream, err: OutputStream, homePath: Path
+      ): Unit = {
+        pingPongAndClose(in, out)
+      }
+
+      override protected def replKeepAliveEvery: FiniteDuration = 10.millis
+
+      def getReplKeepAliveMessage: Message = super.replKeepAliveMessage
```
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329236809/issues/source?bid=3121282&fileBranchId=3206949#l89)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329238132/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329238132/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala (hunk 3 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329238134/issues/source?bid=3121282&fileBranchId=3206949#l89)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 2 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329347417/issues/source?bid=3121282&fileBranchId=3206949#l21)

File: monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala (hunk 1 above)
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/ArchiveSupportTest.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329347417/issues/source?bid=3121282&fileBranchId=3206949#l40)

File: monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      val ab = new Array[Byte](4)
+      in.read(ab)
+      ByteString(ab).utf8String shouldBe "ping"
+      out.write("pong".getBytes)
+      in.close()
+      out.flush()
+      out.close()
+  }
+
+  private[this] def expectString(string: String)(chunks: Seq[Message]) {
```
Comment (lustefaniak) [inline monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala]: Codacy Issue found: [Use : Unit = for procedures](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329347419/issues/source?bid=3121282&fileBranchId=3206949#l46)

File: monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+
+  private[this] def wrongMsg = fail("Response should be BinaryMessage.Strict")
+
+  private[this] val pingPongAndClose: Function2[InputStream, OutputStream, Unit] = {
+    (in, out) =>
+      val ab = new Array[Byte](4)
+      in.read(ab)
+      ByteString(ab).utf8String shouldBe "ping"
```
Comment (pkoryzna) [inline monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala]: we're expecting utf8 strings in the test over here, but it's never coming in from the client as utf8, it's capped at 0xFF
it's not gonna matter for ascii though

**Comment by pkoryzna** [comment]
:sunglasses: :+1:

**Pull request merged**